lib/server/services/ComponentService.js

```diff
@@ -779,28 +820,29 @@ module.exports = function(wilsonConfig, wilsonFrameworkConfig, logger, CacheServ
       //Add behavior dependencies
       _.each(template.dependencies.behaviors, function (behaviorName) {
         try {
-          var curScripts = getBehaviorScripts(behaviorName)
-          behaviorScripts = behaviorScripts.concat(curScripts);
-          serviceScripts = [];
-
-          _.each(behaviorScripts, function (scriptPath) {
-            serviceDependencies = getServiceDependencies(scriptPath, 'behavior');
-
-            //logger.warn('serviceDependencies', serviceDependencies);
-            // Add the scripts of each service dependency
-            _.each(serviceDependencies, function (dependency) {
-              if (serviceLibrary[dependency]) {
-                //logger.warn(dependency);
-                // Get nested dependencies from the service library, and construct the service paths
-                var nestedDependencies = serviceLibrary[dependency].dependencies;
-                var servicePaths = _.pluck(_.pick(serviceLibrary, nestedDependencies), 'path');
-                serviceScripts = serviceScripts.concat(_.uniq(servicePaths));
-                serviceScripts.push(serviceLibrary[dependency].path);
-              }
+          var currBehaviorScripts = buildTimeCache.behaviorScripts[behaviorName] || getBehaviorScripts(behaviorName);
```

Comment:
In general this file need some serious cleanup in order to make these functions smaller and incorporate caching like this to maximize efficiency.